### PR TITLE
Migrated Events app to CBV

### PR DESCRIPTION
**Pull request by amruthasangeeth**
Migrated function based views in Events app to class based views. The following are the views used:
ListView - To list the Events
UpdateView - To edit the Event details
DeleteView - To delete a Event object

### CODE REVIEW

**Comment by tapasweni-pathak** [comment]
@amruthasangeeth Travis is throwing errors, can you please fix them?

```
AIL: test_edit_event_with_invalid_job_date (administrator.tests.test_settings.Settings)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/systers/vms/vms/administrator/tests/test_settings.py", line 689, in test_edit_event_with_invalid_job_date
    self.settings_page)
AssertionError: u'http://localhost:8081/event/list/' == u'http://localhost:8081/event/list/'
```

This failure is present in your other pull requests as well, @smarshy found out that the value should not be same to pass this assertion. 

Let us know if you are not able to resolve the errors.

**Comment by amruthasangeeth** [comment]
@smarshy Can you spot the error it is showing? I am not able to find it.

**Comment by smarshy** [comment]
@amruthasangeeth On editing event, it isn't getting saved. Also, you are having same value retention problem as in the jobs pr. What that means is that if the form gets refreshed / reloaded for some reason ( say invalid values entered) the values that the user has already entered shouldn't vanish.

File: vms/event/forms.py [hunk 1]
```diff
@@ -1,37 +1,36 @@
...
+        end_date = self.cleaned_data.get('end_date')
+
+        if start_date and end_date:
+            if start_date > end_date:
+                msg = u"Start date must be before the end date"
+                self._errors['start_date'] = self.error_class([msg])
+
+        return self.cleaned_data
+
+
```
Comment (smarshy) [inline vms/event/forms.py]: @amruthasangeeth Are there any changes in this model?

File: vms/event/models.py [hunk 2]
```diff
@@ -1,67 +1,68 @@
...
+
+    venue = models.CharField(
+        max_length=30,
+        validators=[
+            RegexValidator(
+                r'^[(A-Z)|(a-z)|(\s)|(\-)|(\')]+$',
+            ),
+        ],
+        blank=True,
+        null=True,
```
Comment (smarshy) [inline vms/event/models.py]: @amruthasangeeth Are there any changes in this model?
I can spot only the autoincrement one

File: vms/event/templates/event/edit.html [hunk 3]
```diff
@@ -1,213 +1,213 @@
...
+                        </div>
+                    </div>
+                {% endif %}
+                <div class="form-group">
+                    <div class="col-md-12 col-md-offset-2">
+                        <button class="btn btn-primary" type="submit">{% trans "Save" %}</button>
+                    </div>
+                </div>
+            </fieldset>
+        </form>
```
Comment (smarshy) [inline vms/event/templates/event/edit.html]: @amruthasangeeth same here, did the template get changed? Tests are very much dependent on the ui structure and the templates

**Comment by tapasweni-pathak** [comment]
@amruthasangeeth Please fix the errors in this and all the consecutive pull requests.

File: vms/event/models.py (hunk 2 above)
Comment (amruthasangeeth) [inline vms/event/models.py]: @smarshy Only the id is the new one.

**Comment by smarshy** [comment]
@amruthasangeeth This has less errors - both are related to edit event. Please try testing this locally. Edit an event with valid details and check if it gets saved.

Also, the second error is due to [this](https://github.com/systers/vms/blob/develop/vms/administrator/tests/test_formFields.py#L226)

Here, since the end date is not wrong, it isn't expected to be erased. Since the same error did not take place for create event, I think this too is related to edit event

**Comment by amruthasangeeth** [comment]
@tapasweni-pathak All the test cases here are passed now :)

File: vms/event/forms.py [hunk 4]
```diff
@@ -1,37 +1,36 @@
...
+class EventDateForm(forms.Form):
+    start_date = forms.DateField(required=True)
+    end_date = forms.DateField(required=True)
```
Comment (tapasweni-pathak) [inline vms/event/forms.py]: @amruthasangeeth What has been changed in this file? Nothing, right? Did you run pep8?

File: vms/event/models.py (hunk 2 above)
Comment (tapasweni-pathak) [inline vms/event/models.py]: @amruthasangeeth Complete file is coming as diff, it is really difficult to review. Can you please undo the unnecessary changes (cleanups, spacing)?

**Comment by tapasweni-pathak** [comment]
@amruthasangeeth Great. :)

If you see you will find that complete files are coming in the diff, it becomes very difficult to review such a diff. Can you please undo the unnecessary changes and only have the useful ones in this pull request? 

You can submit a new pull request for the cleanups, spacing issues e.t.c.

File: vms/event/forms.py (hunk 4 above)
Comment (amruthasangeeth) [inline vms/event/forms.py]: @tapasweni-pathak No changes are made in this file.

File: vms/event/templates/event/edit.html (hunk 3 above)
Comment (tapasweni-pathak) [inline vms/event/templates/event/edit.html]: @amruthasangeeth Did anything changed in this?

File: vms/event/templates/event/delete.html [hunk 5]
```diff
@@ -4,7 +4,7 @@
 
 {% block setting_content %}
     <div class="spacer"></div>
-    <form class="form-horizontal" action="{% url 'event:delete' event_id %}" method="post">
+    <form class="form-horizontal" action="" method="post">
```
Comment (tapasweni-pathak) [inline vms/event/templates/event/delete.html]: @amruthasangeeth Where are we redirecting to delete?

File: vms/event/templates/event/edit.html (hunk 3 above)
Comment (tapasweni-pathak) [inline vms/event/templates/event/edit.html]: @amruthasangeeth Did anything changed here?

File: vms/event/views.py [hunk 6]
```diff
@@ -1,78 +1,88 @@
...
+    def form_valid(self, form):
+        start_date = form.cleaned_data['start_date']
+        if start_date < (datetime.date.today() - datetime.timedelta(days=1)):
+            messages.add_message(self.request, messages.INFO, 'Start date should be today\'s date or later.')
+            return render(self.request, 'event/create.html', {'form': form,})
         else:
-            form = EventForm()
-            return render(request, 'event/create.html', {'form': form, })
-    else:
-        return render(request, 'vms/no_admin_rights.html')
```
Comment (tapasweni-pathak) [inline vms/event/views.py]: @amruthasangeeth We need to redirect to no_admin_rights page in the new code as well, can you please add it back?

Comment (amruthasangeeth) [inline vms/event/views.py]: @tapasweni-pathak That will be taken care of AdministratorLoginRequiredMixin.

File: vms/event/templates/event/delete.html (hunk 5 above)
Comment (amruthasangeeth) [inline vms/event/templates/event/delete.html]: @tapasweni-pathak DeleteView works just like that.

File: vms/event/templates/event/edit.html (hunk 3 above)
Comment (amruthasangeeth) [inline vms/event/templates/event/edit.html]: @tapasweni-pathak No changes are made in this file. May be it came due to the testing I have done locally. Sorry.

File: vms/event/templates/event/delete.html (hunk 5 above)
Comment (tapasweni-pathak) [inline vms/event/templates/event/delete.html]: @amruthasangeeth Then do we need this <form>?

File: vms/event/views.py [hunk 7]
```diff
@@ -82,32 +92,33 @@ def edit(request, event_id):
...
-@login_required
-def list(request):
-    event_list = get_events_ordered_by_name()
-    return render(request, 'event/list.html', {'event_list': event_list})
+class EventListView(LoginRequiredMixin, ListView):
+    model_form = Event
+    template_name = "event/list.html"
+
+    def get_queryset(self):
+        events = Event.objects.all()
```
Comment (tapasweni-pathak) [inline vms/event/views.py]: @amruthasangeeth Show the sorted event list and update this pull request.

File: vms/event/views.py (hunk 6 above)
Comment (tapasweni-pathak) [inline vms/event/views.py]: I have tested all these case manually. @smarshy Can we have few test cases for such cases, where the non admin should be redirected to no_admin_rights page?

File: vms/event/views.py [hunk 8]
```diff
@@ -82,32 +92,33 @@ def edit(request, event_id):
...
-@login_required
-def list(request):
-    event_list = get_events_ordered_by_name()
-    return render(request, 'event/list.html', {'event_list': event_list})
+class EventListView(LoginRequiredMixin, ListView):
+    model_form = Event
+    template_name = "event/list.html"
+
+    def get_queryset(self):
+        events = Event.objects.all().order_by('name')
```
Comment (amruthasangeeth) [inline vms/event/views.py]: @tapasweni-pathak Updated the PR by making the list in sorted order.

Comment (tapasweni-pathak) [inline vms/event/views.py]: @amruthasangeeth Cool.

**Comment by smarshy** [comment]
@tapasweni-pathak There are tests for this, they are commented out as the current vms does not redirect to no admin pages uniformly. I have added a test for this - [here](https://github.com/systers/vms/blob/develop/vms/home/tests/test_functional.py#L93) 
#325 is related to this.
Once all cases are covered, I can uncomment it and check. Or should I do it now itself? The build might fail.

**Comment by tapasweni-pathak** [comment]
@smarshy Okay. Do we have these tests for the cases when it is being redirected to no admin page or all are commented?
